fastify.d.ts

```diff
@@ -102,7 +102,8 @@ declare namespace fastify {
     exposeHeadRoutes?: boolean,
     onProtoPoisoning?: ProtoAction,
     onConstructorPoisoning?: ConstructorAction,
-    logger?: boolean | FastifyLoggerOptions<RawServer> & PinoLoggerOptions | Logger,
+    logger?: boolean | FastifyLoggerOptions<RawServer> & PinoLoggerOptions,
+    loggerInstance?: Logger
```

Comment:
Any reason we're doing this vs could avoid a major semver but I didn't dig into the original issue + config is clear since its a config/options

```typescript
    /** @deprecated */
    logger?: boolean | FastifyLoggerOptions<RawServer> & PinoLoggerOptions,
    loggerOptions?: boolean | FastifyLoggerOptions<RawServer> & PinoLoggerOptions,
    loggerInstance?: Logger
```

Reply:
@jsumners API question: 

Should this be a major change aka deprecate `logger` in favor of more explict `loggerOptions` and `loggerInstance` but I understand the simplicity of `logger` being the standard and `loggerInstance` being a escape hatch for people to pass in an instance.

I also didn't go through the whole issue discussion.

Reply:
https://github.com/fastify/fastify/issues/4624

Reply:
Shorter than I thought ha, didn't see anything but seems to be consensus to add loggerInstance and deprecate init logger object on `logger` 👍

Reply:
FYI I can't resolve but considered: "resolved" 😄